app/models/course/settings/pan_component.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+#
+# This serves as a base class for course settings models that are need settings
+# from more than 1 course component.
+#
+class Course::Settings::PanComponent < SimpleDelegator
+  include ActiveModel::Validations
+
+  def initialize(components)
+    @components = components
+  end
+
+  # Calls the given function from the component settings which respond to the function.
+  #
+  # @param [Symbol] function_name The name of the function to be called.
+  def call_settings_function(function_name)
+    all_settings = settings_interfaces_hash.values.map do |settings|
+      settings.respond_to?(function_name) ? settings.send(function_name) : nil
+    end
+    all_settings.compact.flatten.sort_by { |item| item[:component] }
+  end
+
+  # Calls the given function for updating a setting.
+  # The component key should be passed in the attributes hash.
+  #
+  # @param [Symbol] function_name The name of the function in the Course::Settings::Component
+  #   class which will update the desired setting.
+  # @param [Hash] attributes
+  def call_update_settings_function(function_name, attributes)
```

Comment:
I'm quite uncomfortable with the naming of the two functions. For `call_settings_function`, would prefer something that suggests 1) that a value is being returned, 2) some kind of dispatching/collecting is going on.

Don't have a really good name, but I think something along the lines of `consolidate_components_settings` and `update_component_setting` (singular) might work better.